generator/scripts/generateInputRange.py

```diff
@@ -0,0 +1,97 @@
+import openpyxl  # Read data from Excel
+
+
+# Function to replace single quotes and stringify text
+def replaces_quotes_and_stringify(text):
+    if text is not None:
+        return str(text).replace("'", "\\'")  # Replace single quotes
+    return None
+
+
+# Function to generate inputRange.tsx
+def generate_input_range(input_file: str, output_file: str):
+    try:
+        # Load Excel file
+        workbook = openpyxl.load_workbook(input_file, data_only=True)
+        sheet = workbook["InputRange"]  # Get InputRange sheet
+
+        # Get headers from the first row
+        headers = [cell.value for cell in list(sheet.rows)[0]]
+
+        # Generate TypeScript code
+        ts_code: str = ""  # TypeScript code to be written to file
+        indentation: str = "    "  # 4-space indentation
+
+        # Add imports
+        # ts_code += "import { Labels } from '../types/inputTypes';\n\n"
+
+        # Iterate through each row in the sheet, starting from the second row
+        for row in list(sheet.rows)[1:]:
+            # Create a dictionary from the row values and headers
+            row_dict = dict(zip(headers, (cell.value for cell in row)))
+
+            # Get values from the row dictionary
+            input_range_name = row_dict["inputRangeName"]
```

Comment:
Extra sanitation is probably required here before sending to typescript. The Evolution script should make sure the data entered is good. The one you did is, but to bring those scripts upstream, you should assume the user may make mistakes and you should try to catch as much as possible with nice error messages to help him/her improve the input data. 

What if there are spaces in the name ? You can use some slugify function (there's probably a lib for that in python) to sanitize the input.

Reply:
Yes, my goal is to do automated testing for Generator scripts, It's a good idea to catch the most common errors.